src/core/processing/processingalgorithmparametersmodel.cpp

```diff
@@ -283,6 +286,24 @@ QVariant ProcessingAlgorithmParametersModelBase::data( const QModelIndex &index,
         const QgsProcessingParameterEnum *parameterEnum = dynamic_cast<const QgsProcessingParameterEnum *>( mParameters.at( index.row() ) );
         configuration["options"] = parameterEnum->options();
       }
+      else if ( mParameters.at( index.row() )->type() == QStringLiteral( "source" ) )
+      {
+        QMap<QString, QgsMapLayer *> map = QgsProject::instance()->mapLayers();
+        QVariantList list;
+        QVariantMap supportedLayers;
+        QgsProcessingContext context;
+        context.setProject( QgsProject::instance() );
+        for ( auto it = map.begin(); it != map.end(); ++it )
+        {
+          if ( it.value()->type() == Qgis::LayerType::Vector && QgsProcessingUtils::variantToSource( it.value()->name(), context, mParameters.at( index.row() )->defaultValue() ) )
```

Comment:
@qsavoye , we didn't need the vartiantToSource here, but what we definitively needed was to do a check against the parameter's dataTypes() to filter out of vector layers that are not part of list of geometry types passed by the algorithms. 

I've done that here: https://github.com/opengisch/QField/pull/5690/files#diff-15f7c96b0c450067ace3ed62a83f37fea47213c6cdefbf28bba20747ff6634d2R302